src/error.jl

```diff
@@ -0,0 +1,26 @@
+"Exception type based on PROJ error handling"
+struct PROJError <: Exception
+    msg::String
+    # reset PROJ's error stack on construction
```

Comment:
nit: consider moving this into the line right above `proj_errno_reset(C_NULL)`, so that it is more clearly associated with `function PROJError(msg)` than with `msg::String`